[ExecuTorch] Support bfloat16 in op_index_put

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #5520
* #5519
* __->__ #5500
* #5499

Differential Revision: [D63057744](https://our.internmc.facebook.com/intern/diff/D63057744/)

swolchok committed:
[ExecuTorch] Support bfloat16 in op_index_put



Differential Revision: [D63057744](https://our.internmc.facebook.com/intern/diff/D63057744/)

[ghstack-poisoned]

malfet reviewed (approved):
Do we care about binary size increase of portable ops?

swolchok committed:
Update on "[ExecuTorch] Support bfloat16 in op_index_put"


Differential Revision: [D63057744](https://our.internmc.facebook.com/intern/diff/D63057744/)

[ghstack-poisoned]
